cunumeric/runtime.py

```diff
@@ -351,7 +353,7 @@ def is_supported_type(self, dtype: Union[str, np.dtype[Any]]) -> bool:
 
     def get_numpy_thunk(
         self,
-        obj: Any,
+        obj: Union[ndarray, npt.NDArray[Any]],
```

Comment:
This function should accept anything that can be converted to `numpy.ndarray`, so I think the type should be:
```
obj: Union[ndarray, npt.ArrayLike]
```
This is the best type we can give right now, but the most accurate type would be:
```
obj: Union[SupportsLegateDataInterface, npt.ArrayLike]
```
where `SupportsLegateDataInterface` (or some better name) is a protocol defined in legate.core, that defines a class with a compatible `__legate_data_interface__`.

Classes `cunumeric.ndarray` and `legate.core.Table` both implement this interface.

Reply:
I'll add a protocol for this in a similar legate PR then circle back to use it in another PR in this repo